### DISCUSSION TITLE
Override useFetch/useLazyFetch options (headers) when calling refresh

### DISCUSSION THREAD

**Discussion by zipme** [nuxt/nuxt#17487] [Ideas]
Currently, only the url can be dynamic: ```const { refresh } = useFetch(() => `/something?${id.value}`)```

Sometimes we need to use a new header when refreshing:
```ts
const { locale } = useI18n()

const { data, pending, refresh } = useLazyFetch(
  '/api/xxx,
  {
    headers: { 'Accept-Language': locale.value },
  }
  /* I was hoping we can do this:
  () => ({
    headers: { 'Accept-Language': locale.value },
  })
  */
)
watch(locale, () => {
  refresh()
})
```

**Comment by zipme** [comment]
Alright, found a workaround, so we can do this:
```ts
const { locale } = useI18n()
const { data, pending, refresh } = useLazyFetch('/api/xxx', {
  onRequest: async (context) => {
    context.options.headers = {
      'Accept-Language': locale.value,
    }
  },
})
watch(locale, () => {
  refresh()
})
```

**Comment by ilyaDegtyarenko** [comment]
How can I make it global for useFetch?

**Reply by danya-rnd** [reply]
You cannot use if for global. useFetch doesn't support this feature. The only one way is to write your own composable around $fetch.

**Reply by ilyaDegtyarenko** [reply]
I did just that, only this option has a bug

[https://github.com/nuxt/framework/issues/5238](url)
